infrastructure/cli/pkg/providers/aws/aws.go

```diff
@@ -1,104 +1,578 @@
 package aws
 
 import (
+	"cli/pkg/console"
 	"cli/pkg/kube"
 	"context"
 	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"math/rand"
+	"os"
+	"path"
+	"text/template"
+	"time"
+
 	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
 	"github.com/aws/aws-sdk-go-v2/service/eks"
-	"github.com/aws/aws-sdk-go-v2/service/eks/types"
-	"log"
-	"os"
+	eksTypes "github.com/aws/aws-sdk-go-v2/service/eks/types"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	iamTypes "github.com/aws/aws-sdk-go-v2/service/iam/types"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/mitchellh/go-homedir"
 )
 
+const RolePolicyDocument = `{
+	"Version": "2012-10-17",
+	"Statement": [
+		{
+			"Effect": "Allow",
+			"Principal": {"Service": "eks.amazonaws.com"},
+			"Action": "sts:AssumeRole"
+		},
+		{
+			"Effect": "Allow",
+			"Principal": {"Service": "ec2.amazonaws.com"},
+			"Action": "sts:AssumeRole"
+		}
+	]
+}`
+
+var letters = []rune("abcdefghijklmnopqrstuvwxyz")
+
 type Aws struct {
+	context kube.KubeCtx
 }
 
 func (a *Aws) GetHelmOverrides() []string {
-	return []string{}
+	return []string{"--set", "global.ngrokEnabled=false", "--set", "global.host=http://airy.core"}
 }
 
 func (a *Aws) PostInstallation(namespace string) error {
+	clientset, err := a.context.GetClientSet()
+	if err != nil {
+		return err
+	}
+
+	ingressService, err := clientset.CoreV1().Services("kube-system").Get(context.TODO(), "traefik", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	loadBalancerUrl := ingressService.Status.LoadBalancer.Ingress[0].Hostname
+
+	if err = a.updateIngress("airy-core", loadBalancerUrl, namespace); err != nil {
+		return err
+	}
+	if err = a.updateIngress("airy-core-ui", loadBalancerUrl, namespace); err != nil {
+		return err
+	}
+
+	if err = a.updateHostsConfigMap(loadBalancerUrl, namespace); err != nil {
+		return err
+	}
+
 	return nil
 }
 
+type KubeConfig struct {
+	ClusterName     string
+	EndpointUrl     string
+	CertificateData string
+}
+
+const KubeConfigTemplate = `
+apiVersion: v1
+clusters:
+- cluster:
+    server: {{.EndpointUrl}}
+    certificate-authority-data: {{.CertificateData}}
+  name: kubernetes
+contexts:
+- context:
+    cluster: kubernetes
+    user: aws
+  name: {{.ClusterName}}
+current-context: {{.ClusterName}}
+kind: Config
+preferences: {}
+users:
+- name: aws
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1alpha1
+      command: aws
+      args:
+        - "eks"
+        - "get-token"
+        - "--cluster-name"
+        - "{{.ClusterName}}"
+`
+
 func (a *Aws) Provision() (kube.KubeCtx, error) {
 	cfg, err := config.LoadDefaultConfig(context.TODO())
 	if err != nil {
-		log.Fatal(err)
+		fmt.Println(err.Error())
```

Comment:
The method has an `error` signature, but we treat every error like a hard crash. Imho post-installation errors are expected and can thus be relayed back to the caller.